syntax = "proto3";
import "google/protobuf/descriptor.proto";

message XmlOptions{
    optional string name = 1; // field or message name for xml tag <name></name>
    optional string description = 2;
    optional bool is_element = 3; // it true <name> otherwise attribute <parent name= ...>
    optional bool can_empty = 4; // if true can <name/> or <parent>
                                 // if false <name></name> or <parent name="">
}

extend google.protobuf.MessageOptions{
    optional XmlOptions xml_message_options = 55000;
}

extend google.protobuf.FieldOptions{
    optional XmlOptions xml_field_options = 55001;
}

extend google.protobuf.EnumOptions{
    optional XmlOptions xml_enum_options = 55002;
}

extend google.protobuf.EnumValueOptions{
    optional XmlOptions xml_enum_value_options = 55003;
}
